app/_plugins/blocks/konnect_crd.rb

```diff
@@ -44,16 +44,22 @@ def render(context) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
     end
 
     def add_defaults(config)
+      kind = config['kind']
+
       defaults = {
         'kind' => '@TODO', # Needed to make sure kind is the first item in the YAML output
         'apiVersion' => 'konnect.konghq.com/v1alpha1',
         'metadata' => {
           'name' => '@TODO',
           'namespace' => 'kong'
-        },
-        'spec' => {}
+        }
       }
 
+      # Only include 'spec: {}' for non-KongPlugin resources
+      unless kind == 'KongPlugin'
+        defaults['spec'] = {}
+      end
```

Comment:
this would be something like:

```ruby
      if  config['kind'] == 'KongPlugin'
        defaults.delete('spec') 
      end
```

and you can also remove `kind = config['kind']` from the top of the method